internal_types/src/schema.rs

```
@@ -937,4 +1033,81 @@ mod test {
             InfluxColumnType::Timestamp
         );
     }
+
+    #[test]
+    fn test_sort() {
+        let mut sort_key = SortKey::with_capacity(3);
+        sort_key.push("tag4", Default::default());
+        sort_key.push("tag3", Default::default());
+        sort_key.push("tag2", Default::default());
+        sort_key.push("tag1", Default::default());
+        sort_key.push(TIME_COLUMN_NAME, Default::default());
+
+        let schema1 = SchemaBuilder::new()
+            .influx_field("the_field", String)
+            .tag("tag1")
+            .tag("tag2")
+            .tag("tag3")
+            .tag("tag4")
+            .timestamp()
+            .measurement("the_measurement")
+            .build()
+            .unwrap()
+            .with_sort_key(&sort_key);
+
+        let projected = schema1.project(
+            schema1
+                .select(&["tag4", "tag2", "tag3", "time"])
+                .unwrap()
+                .as_slice(),
+        );
+
+        let projected_key: Vec<_> = projected.sort_key().unwrap().iter().map(|x| *x.0).collect();
+
+        let m1 = SchemaMerger::new().merge(&schema1).unwrap().build();
+
+        let m2 = SchemaMerger::new()
+            .merge(&schema1)
+            .unwrap()
+            .build_with_sort_key(&sort_key);
+
+        assert_eq!(schema1.sort_key().unwrap(), sort_key);
+        assert_eq!(m1.sort_key(), None);
+        assert_eq!(m2.sort_key().unwrap(), sort_key);
+        assert_eq!(projected_key, vec!["tag4", "tag3", "tag2"])
+    }
+
+    #[test]
+    fn test_sort_missing_column() {
```

I was unsure what the correct behaviour should be if you specify a sort key containing columns that are missing. Schema::sort_key() will only return the prefix for which it has all the columns and so it seemed reasonable to just ignore missing columns, but I'm not sure...

---

I would personally prefer an `Err` or `panic!` -- the rationale being that silently ignoring a missing column may mask a bug and make it much harder to track down.

---

This now returns errors